bindings/pybind11/idyntree_core.cpp

```diff
@@ -179,35 +49,27 @@ void transformClassDefinition(py::class_<Transform>& transform) {
 }  // namespace
 
 void iDynTreeCoreBindings(pybind11::module& module) {
-  // Vectors and matrices.
-  py::class_<VectorDynSize> vector_dyn(module, "VectorDynSize",
-                                       py::buffer_protocol());
-  vectorDynSizeClassDefinition(vector_dyn);
-  createFixSizeVector<3>(module, "Vector3");
-  createFixSizeVector<4>(module, "Vector4");
-  createFixSizeVector<6>(module, "Vector6");
-
-  py::class_<MatrixDynSize> matrix_dyn(module, "MatrixDynSize",
-                                       py::buffer_protocol());
-  matrixDynSizeClassDefinition(matrix_dyn);
-  createFixSizeMatrix<3, 3>(module, "Matrix3x3");
-  createFixSizeMatrix<4, 4>(module, "Matrix4x4");
-  createFixSizeMatrix<6, 6>(module, "Matrix6x6");
 
   // Positions, Rotations and Transforms.
-  py::class_<PositionRaw, VectorFixSize<3>>(module, "_PositionRaw")
+  py::class_<PositionRaw>(module, "_PositionRaw")
       // Do not expose constructor as we do not want users to use this class.
       .def("__repr__", &PositionRaw::toString);
 
+
   py::class_<Position, PositionRaw>(module, "Position")
       .def(py::init())
       .def(py::init<double, double, double>())
+      .def(py::init([](const VectorFixSize<3>& position)
+                    {
+                        return Position(position(0), position(1), position(2));
+                    }))
       .def(py::self + py::self)
       .def(py::self - py::self)
       .def(-py::self)
       .def_static("Zero", &Position::Zero);
+  py::implicitly_convertible<iDynTree::VectorFixSize<3>, iDynTree::Position>();
```

Comment:
```python
position = Position([0, 1, 3])
```
should work but
```python
transform = Transform(np.array([[1,0,0],[0,1,0],[0,0,1]]), [0,0,0])
```
won't.

Without the explicit caster you need to do something like this:
```python
transform = Transform(Rotation(np.array([[1,0,0],[0,1,0],[0,0,1]])), Position([0,0,0]))
```
This is in general valid for each function that takes as input a `position` or a `Rotation`

Reply:
Yes. This is what I was referring to.

I like that you can pass a Python list instead of an iDynTree vector, but I am afraid of doing this silently for all other objects that can be represented by a vector but have some other meanings, e.g. Rotation